source/glkernel/include/glkernel/sample.hpp

```diff
@@ -207,6 +208,33 @@ size_t poisson_square(tkernel<glm::tvec2<T, P>> & kernel, const T min_dist, cons
     return k + 1;
 }
 
+template <typename T, glm::precision P>
+size_t n_rooks(tkernel<glm::tvec2<T, P>> & kernel)
+{
+	assert(kernel.depth() == 1);
+	assert(kernel.width() == kernel.height());
+
+	auto stratum_size = 1.0 / kernel.width();
+	std::random_device RD;
+	std::mt19937_64 generator(RD());
+	std::uniform_real_distribution<> jitter_dist(0.0, stratum_size);
+
+	std::vector<unsigned int> pool;
```

Comment:
comment this: e.g., "Pool of row values, column shuffling is done independently, see below"